### DISCUSSION TITLE
Can TimeSeriesForestClassifier support multivariate series? I have multiple values at each point in time that I want to use in classification.

### DISCUSSION THREAD

**Discussion by TannerCypret** [alan-turing-institute/sktime#7232] [Q&A] [answered]
Can this be done by adding another regressor? I have multiple values at each time step and when I pass that in I get the error: 

`Data seen by timeSeriesForestClassifier instance has multivariate series, but this TimeSeriesForestClassifier instance cannot handle multivariate series. Calls with multivariate series may result in error or unreliable results.`

Is there a way to support multiple values at each time step?

**edit: I was able to follow this tutorial for multivariate time series using the Basic Motions data set: https://medium.com/swlh/using-sktime-for-time-series-classification-part-2-b07bb6b56d5**

However, my time series data is of unequal length and I get this error now:

`Data seen by TimeSeriesForestClassifier instance has unequal length series, but this TimeSeriesForestClassifier instance cannot handle unequal length series. Calls with unequal length series may result in error or unreliable results.`

Is there a way to support unequal length data? I might see about truncating what I have for now so that they are the same length.

**Comment by fkiraly** [comment, marked as answer]
@TannerCypret, yes, for this you need to ensure you use a *time series distance that is capable of handling unequal length time series*, and build it manually.

The tag for this is `capability:unequal_length`, you can find these programmatically via the `all_estimators` utility (using `transformer-pairwise` as the type). You can also build them out of simpler components. You also want to ensure that you have `capability:multivariate`, you can build a multivariate distance out of any univariate one using `IndepDist`.

DTW is a common choice, and out of the four implementations available in `sktime`, the following support `capability:unequal_length`: `DtwDtaidistMultiv`, `DtwPythonDist`, and `DtwDistTslearn`, but not the one by the Bagnall group.

The different implementations differ slightly by python dependencies, performance, parameters.

**Comment by fkiraly** [comment]
@TannerCypret, I have now updated docstrings of the knn estimtaors in `sktime` - hope the new docstring makes it clearer how to achieve what you wanted: https://github.com/sktime/sktime/pull/7241
